server/src/main/java/org/elasticsearch/search/fetch/subphase/FetchSourceContext.java

```diff
@@ -84,12 +121,48 @@ public String[] excludes() {
         return this.excludes;
     }
 
+    public boolean includeVectors() {
+        return this.includeVectors;
+    }
+
     public boolean hasFilter() {
-        return this.includes.length > 0 || this.excludes.length > 0;
+        return this.includes.length > 0 || this.excludes.length > 0 || this.includeVectors == Boolean.FALSE;
```

Comment:
I don't think we should roll in `includesVectors` checking into `hasFilter`. If you look at where `hasFilter` is used, it is sometimes used to answer the question "are there user-defined filters?". Since we are excluding vectors by default in most code paths, vector exclusions are not user defined.

Let's remove the `includesVectors` check here and instead evaluate each `hasFilter` usage and determine if we also need to check `includesVectors`.